src/Repository/UserRepository.php

```diff
@@ -215,4 +216,73 @@ public function countUserByStatus(?Territory $territory = null, ?User $user = nu
 
         return $qb->getQuery()->getOneOrNullResult();
     }
+
+    public function findExpiredUsers(): array
+    {
+        $limitConservation = '3 years'; // <- durée légale de conservation des données (ou rendre anonyme)
+        // $limitConservation = '1 years'; //pour test
+        $dateLimit = new \DateTime('-'.$limitConservation);
+
+        // retourne les utilisateurs :
+        // - non connectés depuis plus de limitConservation
+        // - et n'etant pas occupant ou declarant sur des signalements actif (creation/edition/dernier suivi) depuis limitConservation
+        $qb = $this->createQueryBuilder('u')
+            ->select('u')
+            ->leftJoin('u.signalementUsagerDeclarants', 'sud')
+            ->leftJoin('u.signalementUsagerOccupants', 'suo')
+            ->leftJoin('sud.signalement', 'sd')
+            ->leftJoin('suo.signalement', 'so')
+
+            ->where('(u.lastLoginAt < :dateLimit) OR (u.lastLoginAt IS NULL AND u.createdAt < :dateLimit)')
+            ->andWhere('sd.createdAt < :dateLimit AND so.createdAt < :dateLimit')
+            ->andWhere('(sd.modifiedAt IS NULL OR sd.modifiedAt < :dateLimit) AND (so.modifiedAt IS NULL OR so.modifiedAt < :dateLimit)')
+            ->andWhere('(sd.lastSuiviAt IS NULL OR sd.lastSuiviAt < :dateLimit) AND (so.lastSuiviAt IS NULL OR so.lastSuiviAt < :dateLimit)')
+
+            ->setParameter('dateLimit', $dateLimit);
+
+        return $qb->getQuery()->execute();
+    }
+
+    public function findInactiveUsers(?bool $isArchivingScheduled = null, ?DateTime $archivingScheduledAt = null): array
+    {
+        $limitConservation = '11 months';
+        $dateLimit = new \DateTime('-'.$limitConservation);
+
+        $qb = $this->createQueryBuilder('u')
+            ->select('u', 'p', 't')
+            ->leftJoin('u.partner', 'p')
+            ->leftJoin('u.territory', 't')
+
+            ->where('u.lastLoginAt IS NOT NULL AND u.lastLoginAt < :dateLimit')
+            ->andWhere('u.roles != :roles')
+            ->andWhere('u.statut != :statut')
+
+            ->setParameter('dateLimit', $dateLimit)
+            ->setParameter('roles', '["ROLE_USAGER"]')
+            ->setParameter('statut', User::STATUS_ARCHIVE);
+
+        if (true === $isArchivingScheduled) {
+            $qb->andWhere('u.archivingScheduledAt IS NOT NULL');
+        }
+        if (false === $isArchivingScheduled) {
+            $qb->andWhere('u.archivingScheduledAt IS NULL');
+        }
+
+        if ($archivingScheduledAt) {
+            $qb->andWhere('u.archivingScheduledAt = :date')
+                ->setParameter('date', $archivingScheduledAt->format('Y-m-d'));
+        }
+
+        return $qb->getQuery()->execute();
```

Comment:
Dans le projet pour les Select on utilise plutôt  `$qb->getQuery()->getResult();` et ses équivalents

Reply:
C'est ce que j'ai toujours utiliser mais j'ai remarqué qu'à présent dans la doc Symfony https://symfony.com/doc/current/doctrine.html#querying-with-the-query-builder ils utilisent le execute.

Suite à ta remarque j'ai vérifié et le getResult en interne fait appel au execute : 
![getResult](https://github.com/MTES-MCT/histologe/assets/7130780/78bb1db3-8c71-4168-9b67-25bc057ed631)

Je sent venir la dépréciation donc je pense qu'on peux commencer à changer nos habitudes